File: strategy_pattern/demo/ducks/red_head_duck.py
from ducks.duck import Duck
from interfaces.fly_behaviour import FlyBehaviour
from interfaces.quack_behaviour import QuackBehaviour

class RedHeadDuck(Duck):
    def __init__(self, fly_behaviour: FlyBehaviour, quack_behaviour: QuackBehaviour):
        super().__init__(fly_behaviour, quack_behaviour)

    def display(self)-> str:
        return "Looks like a Red Head!"

File: strategy_pattern/java/demo/RedheadDuck.java
package strategy_pattern.java.demo;

public class RedheadDuck extends Duck{

    public RedheadDuck(){

        flyBehaviour = new FlyNoWay();
        quackBehaviour = new Quack();
    }

    @Override
    public void display() {
        
        System.out.println("I'm a Redhead duck");
    }
    
}
